--- search.c.orig	2021-09-24 07:29:05.898135979 +1000
+++ search.c	2021-09-24 07:59:42.995002898 +1000
@@ -161,6 +161,26 @@ isearch(int dir)
 			 * The same issue can also be observed in uEmacs/Pk
 			 */
 
+			/*
+			 * BERTO. I never use "ESC" to exit from search.
+			 * I always use either "Enter" or an arrow key.
+			 *
+			 * pEmacs spits out garbage like [B or [A if you
+			 * hit an arrow key to exit from search.
+			 * 
+			 * The following kludge hack "absorbs" the garbage 
+			 * that gets spat out if you try to use an arrow to 
+			 * exit from search, but it means that if you actually
+			 * use "ESC" to exit search then you have to type two 
+			 * more characters to really exit search. 
+			 *
+			 * Not everyone likes this behavior so only apply
+			 * this patch to pEmacs if you're like me and never 
+			 * use just the ESC key to exit from search.
+			 */
+			ttgetc();
+			ttgetc();
+			
 			/* FALLTHRU */
 		case CTRL_M:
 			srch_lastdir = dir;
