--- main.c.orig	2021-09-24 17:51:50.109115639 +1000
+++ main.c	2021-09-24 18:14:23.456919036 +1000
@@ -31,6 +31,7 @@ extern int anycb ();
 extern BUFFER *bfind ();
 extern int ttgetc();
 extern int rdonly();
+extern int filesave (int f, int n);
 
 int main (int argc, char *argv[]);
 void edinit (char bname[]);
@@ -266,7 +267,7 @@ int getctl ()
 int quit (int f, int n)
 {
   int s;
-
+#ifdef BERTO
   if (f != FALSE	       /* Argument forces it */
       || anycb () == FALSE     /* All buffers clean */
       || (s = mlyesno ("Modified buffers exist. Leave anyway")) == TRUE)
@@ -276,6 +277,25 @@ int quit (int f, int n)
     }
   mlwrite ("");
   return (s);
+#endif
+  if (f == FALSE && anycb() == TRUE) {
+	  s = mlyesno ("Modified buffers exist. Save");
+	  if (s == TRUE) {
+		  filesave (0, 0);
+		  goto quit_exit;
+	  } else {
+		  s = mlyesno ("Modified buffers exist. Leave anyway");
+		  if (s == TRUE) {
+			  goto quit_exit;
+		  }
+	  }
+	  mlwrite ("");
+	  return (s);
+  }
+quit_exit:
+  vttidy ();
+  exit (0);
+  extern int filesave (int f, int n);
 }
 
 /*
